### PR TITLE
Fix Delta.arg_constraints and .support

### PR DESCRIPTION
**Pull request by fritzo**
Ports https://github.com/pyro-ppl/numpyro/pull/902 to Pyro.

Note Pyro does not use `support.is_discrete`, and `dependent_constraint` will not support `is_discrete` until #2753 .

### CODE REVIEW

**Comment by fehiepsi** [comment]
I guess this is blocked by  #2753 to make `constraints.independent` available.

**Comment by fritzo** [comment]
@fehiepsi we already have an `IndependentConstraint` in Pyro dev, so I've used that to unblock this PR.

**Review by fehiepsi — APPROVED** [review]
LGTM.

**Pull request merged**
